src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java

```diff
@@ -488,6 +490,25 @@ public void dumpAndRestore() {
     assertTrue(jedis.exists("foo2"));
   }
 
+  @Test(expected = JedisBusyException.class)
+  public void dumpAndRestoreDup() {
+    jedis.set("foo1", "bar1");
+    byte[] sv = jedis.dump("foo1");
+    jedis.restore("foo1", 0, sv);
+    assertTrue(jedis.exists("foo1"));
+  }
+
+  @Test
+  public void dumpAndRestoreReplace() {
```

Comment:
copy and keys params were used by MIGRATE command , but i cannot find any unit tests for migrate command , it seems to be that migrate command tests need at least two redis server and the target server require nopass.